### PR TITLE
Tickets/preops 3763

### PR DESCRIPTION
**Pull request by emanehab99**
Tests were failing because for some reason some bokeh models are owned by multiple documents. This happens when scheduler dashboard and prenight dashboard both run in tests. Resetting bokeh output before and after tests fixes the issue

### CODE REVIEW

File: tests/test_scheduler_dashboard.py [hunk 1]
```diff
@@ -66,17 +65,24 @@ class TestSchedulerDashboard(unittest.TestCase):
...
+        sched_app_bokeh_model = sched_app.get_root()
+        with TemporaryDirectory() as scheduler_dir:
+            sched_out_path = Path(scheduler_dir)
+            sched_saved_html_fname = sched_out_path.joinpath("sched_test_page.html")
+            bokeh.plotting.output_file(filename=sched_saved_html_fname, title="Scheduler Test Page")
+            bokeh.plotting.save(sched_app_bokeh_model)
+        bokeh.io.reset_output()
 
     def test_read_scheduler(self):
+        print(TEST_PICKLE)
```
Comment (ehneilsen) [inline tests/test_scheduler_dashboard.py:82]: This print is probably no longer necessary (or logging would be better).

File: schedview/app/scheduler_dashboard/scheduler_dashboard.py [hunk 2]
```diff
@@ -73,6 +73,11 @@
 """
 
 
+def get_color_palettes():
```
Comment (ehneilsen) [inline schedview/app/scheduler_dashboard/scheduler_dashboard.py:76]: Consider either indicating that this private (preceded by and underscore), or adding a docstring (if it is not).

File: schedview/app/scheduler_dashboard/scheduler_dashboard.py [hunk 3]
```diff
@@ -224,9 +230,11 @@ def _update_date(self):
             )
         ).mjd
 
+        print(f"date in mjd: {self._date_time}")
```
Comment (ehneilsen) [inline schedview/app/scheduler_dashboard/scheduler_dashboard.py:233]: Is this print still necessary (or should it be replaced by a log message)?

File: schedview/app/scheduler_dashboard/scheduler_dashboard.py [hunk 4]
```diff
@@ -465,7 +473,9 @@ def read_scheduler(self):
             return False
 
     def make_scheduler_summary_df(self):
-        """Update conditions, make the reward and scheduler summary dataframes.
+        """
```
Comment (ehneilsen) [inline schedview/app/scheduler_dashboard/scheduler_dashboard.py:476]: The summary line for a docstring should be on the same line as the initial tripple quote (see https://developer.lsst.io/python/numpydoc.html , in the section whose heading starts with "Docstrings SHOULD begin with".

**Review by ehneilsen — CHANGES_REQUESTED** [review]
Mostly, this looks good! I just have a few minor stylistic requests.

First, there are some public functions that do not have docstrings. In general, a function should either have a docstring, or marked as private (have a preceeding underscore), or both.

Second, some of the docstrings that are there have their summary sentences on a different line from the opening tripple quote. This messes up some of the doc extraction tools. See https://developer.lsst.io/python/numpydoc.html . 

Finally, there are still a few extraneous "print" statements in there, obviously included for debugging. If they should be there long term, they should probably be logging messages instead.

In each of these cases, I just marked one example with a comment, but there are multiple instances of each.

**Comment by ehneilsen** [comment]
Can you do a rebase onto main and resolve the resultant merge conflicts?

**Comment by emanehab99** [comment]
@ehneilsen Does that look good now? Could it be merged?

**Comment by emanehab99** [comment]
> Can you do a rebase onto main and resolve the resultant merge conflicts?

I did. Please let me know if I need to do something different

**Comment by emanehab99** [comment]
> Can you do a rebase onto main and resolve the resultant merge conflicts?
Comments addressed. It should be up to date now.

**Comment by emanehab99** [comment]
I'm not sure what the ticks and crosses next to commits mean. That's what I did to rebase the branch
```
git checkout main
git pull 
git checkout tickets/PREOPS-3763
git rebase main
git pull
git push
```

**Comment by timj** [comment]
You made one mistake in:

```
git checkout main
git pull 
git checkout tickets/PREOPS-3763
git rebase main
git pull
git push
```
It should be:
```
git checkout main
git pull 
git checkout tickets/PREOPS-3763
git rebase main
git push -f
```

That `git pull` at the end forces a redownload of the original ticket branch and git makes a merge commit to deal with them being different commits because you had rebased locally.

The `-f` tells git that the local version is the proper one.

**Comment by emanehab99** [comment]
> You made one mistake in:
> 
> ```
> git checkout main
> git pull 
> git checkout tickets/PREOPS-3763
> git rebase main
> git pull
> git push
> ```
> 
> It should be:
> 
> ```
> git checkout main
> git pull 
> git checkout tickets/PREOPS-3763
> git rebase main
> git push -f
> ```
> 
> That `git pull` at the end forces a redownload of the original ticket branch and git makes a merge commit to deal with them being different commits because you had rebased locally.
> 
> The `-f` tells git that the local version is the proper one.


Thank you. I've done that now but I'm not happy with the commit history (repeating my commits multiple times). Is there a way to clean it?

**Comment by timj** [comment]
You can do `git rebase -i main` and squash related commits.

**Comment by emanehab99** [comment]
> You can do `git rebase -i main` and squash related commits.


That worked! Looks much better. Thank you so much :)

**Review by ehneilsen — APPROVED** [review]

**Pull request merged**